src/node.jl

```diff
@@ -1,3 +1,14 @@
+
+Base.@kwdef mutable struct NodeOptions
+    rate::Float64 = 100
+end
+
+Base.@kwdef mutable struct NodeFlags
+    did_error::Bool = false
+    is_running::Threads.Atomic{Bool} = Threads.Atomic{Bool}(false)
+    should_finish::Threads.Atomic{Bool} = Threads.Atomic{Bool}(false)
```

Comment:
Why do you have to use the `Threads.Atomic` for `is_running` and `should_finish` but not `did_error`? You set the value of `did_error` before closing the nodes task.

Reply:
I could use an atomic, but there's no contention for `did_error`, since it's a flag that gets set before the thread closes and only intended to be read after the node closes.

Reply:
I ended up changing it to be consistent.